docs/app/writing-tests.md

```diff
@@ -0,0 +1,112 @@
+# Writing Tests
+
+[pytest](https://docs.pytest.org) is our test runner, which is simple but
+powerful. If you are new to pytest, reading up on how [fixtures
+work](https://docs.pytest.org/en/latest/explanation/fixtures.html) in particular might be
+helpful as it's one area that is a bit different than is common with other
+runners (and languages).
+
+## Naming
+
+pytest automatically discovers tests by [following a number of
+conventions](https://docs.pytest.org/en/stable/goodpractices.html#conventions-for-python-test-discovery)
+(what it calls "collection").
+
+For this project specifically:
+
+- All tests live under `app/tests/`
+- Under `tests/`, the organization mirrors the source code structure
+  - The tests for `app/api/route/` are found at `app/test/api/route/`
+- Create `__init__.py` files for each directory. This helps [avoid name
+  conflicts when pytest is resolving
+  tests](https://docs.pytest.org/en/stable/goodpractices.html#tests-outside-application-code).
+- Test files should begin with the `test_` prefix, followed by the module the
+  tests cover, for example, a file `foo.py` will have tests in a file
+  `test_foo.py`.
+- Test cases should begin with the `test_` prefix, followed by the function it's
+  testing and some description of what about the function it is testing.
+  - In `tests/api/test_example.py`, the `test_example_patch_404` function is a test
+    (because it begins with `test_`), that covers the `users_patch` function's
+    behavior around 404 responses.
```

Comment:
Can we use an example from something like `tests/api/route/test_eligibility.py`?